operator/v1/types_ingress.go

```diff
@@ -1382,6 +1382,110 @@ type IngressControllerHTTPHeaders struct {
 	// +nullable
 	// +optional
 	HeaderNameCaseAdjustments []IngressControllerHTTPHeaderNameCaseAdjustment `json:"headerNameCaseAdjustments,omitempty"`
+
+	// actions specifies options for performing action on headers.
+	// Note that this option only applies to cleartext HTTP connections
+	// and to secure HTTP connections for which the ingress controller
+	// terminates encryption (that is, edge-terminated or reencrypt
+	// connections).  Headers cannot be set or deleted for TLS passthrough
+	// connections.
+	// Setting HSTS header is supported to be set via route annotation haproxy.router.openshift.io/hsts_header and not via this API.
+	// Setting httpCaptureHeaders for the headers set via this API will not work.
+	// If spec.clientTLS, spec.httpHeaders.forwardedHeaderPolicy, spec.httpHeaders.uniqueId, spec.httpHeaders.headerNameCaseAdjustments is set
+	// and if you set the same header via this API then that header will get replaced with the value specified via this API.
+	// If cache-control is set using this API then the existing value of cache-control will get replaced.
+	// +optional
+	Actions IngressControllerHTTPHeaderActions `json:"actions,omitempty"`
```

Comment:
I could be wrong, but I think you need to make this a pointer. See comment in https://github.com/openshift/cluster-ingress-operator/pull/872/files#r1109108270

Reply:
No we don't need it to be a pointer. The api server will check for it.

Reply:
Using a pointer type reduces visibility. Using a non-pointer type means that the API server always includes the field when it generates json/yaml for an object.

Reply:
The Ingress object is a configuration API, where the Route is a workload API, they have different rules for pointers, but this one is correct